src/openssl/ssl/socket.cr

```diff
@@ -1,4 +1,56 @@
-class OpenSSL::SSL::Socket
+abstract class OpenSSL::SSL::Socket
+  class Client < Socket
+    def initialize(io, context : Context::Client = Context::Client.new, sync_close : Bool = false, hostname : String? = nil)
+      super(io, context.as(Context), sync_close)
```

Comment:
I don't get any error if I leave that case out. What's a code to reproduce the bug?